### ISSUE TITLE
The built-in client hangs when using tty transfer

### ISSUE THREAD

**Issue by holyjak** [nrepl/nrepl#126]
I am sure it is me who does something wrong, but I'd like to understand what and 

## Expected behavior

Starting the server with the tty transfer:

```clj
;; $  clj -R:nrepl
user=> (require '[nrepl.transport :as transport])
nil
user=> (require '[nrepl.server :refer [start-server stop-server]])
nil
user=> (start-server :port 55555 :transport-fn transport/tty)
```

and connecting with the client using the same transfer:

```sh
$ clj -R:nrepl -m nrepl.cmdline --connect --host 127.0.0.1 --port 55555 -t nrepl.transport/tty
```

I expect to see the prompt and be able to interact with it.

## Actual behavior

The client blocks, there is no output and it has to be ^C stopped. At that moment, the server prints `Exception in thread "nREPL-worker-0" java.lang.Error: java.net.SocketException: Socket closed ... at nrepl.transport$tty$write__427.invoke(transport.clj:136)`.

Beware: I see the same behavior when I omit the `-t nrepl.transport/tty` option so it seems it somehow is not processed correctly?

### Clojure version

1.10.0

### Java version

1.8

### Operating system

OSX

### nREPL

0.5.3

**Comment by shen-tian** [comment]
Can confirm that this hangs:

`clj -Sdeps '{:deps {nrepl {:mvn/version "0.6.0"}}}' -m nrepl.cmdline --transport nrepl.transport/tty --interactive`

So this has been broken for a while now.

Looking at the changelog, the `--transport` flag was only fixed in `0.6.0`, so not sure there was ever a command line version of this that worked.

Gonna do some digging on when/how this last worked, but suspect this feature is pretty end of line, given the Socket REPL? Do we know anyone that uses this?

**Comment by bbatsov** [comment]
I think your suggestion is quite reasonable. It's quite unlikely someone will want to use a `tty` server with the default client.
